File: producer/go/dispatcher.go
package producer

import (
	"encoding/json"
)

func Dispatcher(str string) string {
	flow := Compile(str)
	switch flow.Entity {
	case "Book":
		book := NewBook()
		for _, call := range flow.Calls {
			switch call.FunctionName {
			case "target":
				book.Target()
			case "source":
				book.Source(call.Parameters[0].TypeValue)
			}
		}

		out, _ := json.Marshal(book)
		return string(out)
	default:
		return ""
	}
}

File: producer/js/src/dispatcher.ts
import {Compile} from "./ast";
import {Book} from "./book";

export function Dispatcher(str: string) {
    let flow = Compile(str)
    switch (flow.Entity) {
        case "Book":
            let book = new Book();
            for (let call of flow.Calls) {
                switch (call.FunctionName) {
                    case "Target":
                        book.target()
                        break;
                    case "Source":
                        book.source(call.Parameters[0].TypeValue)
                        break;
                }
            }

            return JSON.stringify(book)
        default:
            return ""
    }
}
